c_binding/utils.c

```diff
@@ -114,3 +114,19 @@ int _read_file(const char *path, uint8_t *buff, ssize_t *size, size_t max_size)
     }
     return rc;
 }
+
+/* Find the first trailing space character,
+ * insert a NULL character in its place.
+ */
+void _trim_trailing_spaces(char *string_ptr)
+{
```

Comment:
If `string_prt` is not terminated by `\0`[1], we may get a segfault here. Please consider to include a string length here.

[1]: I may happen on some buggy kernel driver or firmware, and SCSI standard said it **may** contain one or more `\0` as terminator